[WIP] frontend: 5xx page redesign

vaidap opened the pull request:
This is a continuation of #4375 - i had to close it when i was cleaning up my git branches as I had not taken the right approach.

Changes made:
- CSS is fixed
- moved images to `static/images/`
- added the fish image at the bottom
![screen shot 2017-05-30 at 15 21 45](https://cloud.githubusercontent.com/assets/13666710/26587934/bcb53f02-454b-11e7-86f3-6385a64f9038.png)

Let me know if I should make any further changes.

@YagoGG would you mind taking a look at this?

smarx commented:
*Automated message from Dropbox CLA bot*

@vaidap, it looks like you've already signed the Dropbox CLA. Thanks!

vaidap commented:
- added a nice shadow effect
- changed the background colour to make it more vibrant
- removed italics from large text

timabbott commented:
Thanks for working o this @vaidap!  I took a look, and I have a few suggestions:
* This white outline at the top here seems like a bug:
![image](https://cloud.githubusercontent.com/assets/2746074/26603185/cfca7c58-453a-11e7-9770-3c7a4b0ca347.png)
* The "Log In" link in the corner we should probably just remove, since it's not going to work on this 500 page, which happens when the whole Zulip server is down (we'll want links, though, for the other 500 pages, which involve a single page having 500ed).
* Since the bubbles have outlines, the fish probably should too
* The grey spots don't look good in the places where they overlap the boundary of the tentacles:
![image](https://cloud.githubusercontent.com/assets/2746074/26603295/29cc53f2-453b-11e7-8acb-09e4ca1be681.png); maybe make them smaller?
* Probably the 500 should go away and it should just say "Internal server error" much bigger, since most users don't know what a "500" is.
* We probably should get rid of the word "Zulip" being a link, though it might make sense to add a Zulip
* We actually intentionally removed the Twitter/etc. links from this page, since the links are inaccurate for self-hosted Zulip servers.  For the other 500 page and the 404, where we do have access to variables, we will can still use those links, but for this version of the 500 page, we should probably just have some brief text letting them know that the page will reload soon after service is restored (or something).

Hmm, that's more comments than I'd intended to leave, but I think with these tweaks this should be basically finished, and then we'll be able to do the other error pages much more quickly.  Thanks @vaidap!

YagoGG commented:
Great job, @vaidap! Looks nice.

Here are a few comments, in addition to Tim's:

- I can't see the font you show in your screenshot, for me it's using Arial. I'm using Chrome on Ubuntu.
- I'd use the same font in the `.errorname` header as well, or at least for the "internal server error" (maybe bolder?).
- The white border shown in Tim's first screenshot doesn't appear in my dev env. Maybe we should look into that?

The rest looks awesome, I'm looking forward to see this merged :)

YagoGG commented:
On a side note, I don't fully agree with removing the 500 error code. Even though most users don't know what it is, it's something relatively useful for people with technical profiles (especially taking into account that a lot of these pages are probably going to be reported to IT departments).

As an example from other services, you can still see the HTTP error code if you get one in a [Google page](http://google.com/foo).

If the aim is to make it less important visually, maybe we could make it smaller, or match the size of the "internal server error" message. What do you think, @timabbott?

timabbott commented:
Yeah, we can try making it the same size as the human-readable message.  I mostly just don't think it should have more prominence than the text.

vaidap commented:
- implemented all of Tim's changes (except white outline)
- changed the sizes of 500/internal server error so that they match
- for the white outline, Tim, what browser are you using? This might have something to do with the `box-shadow` property.
- as for fonts, I haven’t changed the defaults— seems like it is all set in body, both the header and the main text font: `font-family: 'Humbug', Helvetica, Arial, sans-serif;`. Should I make changes to this? I added `lead` to both h1's so they are bolded.
this is what it looks like now for me: 
![screen shot 2017-05-31 at 15 35 08](https://cloud.githubusercontent.com/assets/13666710/26637629/df2dc9ce-4617-11e7-8725-62434087b94a.png)
it is starting to properly come together! 🎉

vaidap commented:
* changed up the text so the header reads `500: Internal server error`, and there is no larger text inside the body, to reflect #5314.

YagoGG commented:
I've been doing some further testing and looks like Humbug isn't working for me anywhere, so it isn't related to your PR.

One small thing I noticed: shouldn't there be shadow over those fish in the bottom as well?

![image](https://user-images.githubusercontent.com/7356565/26984772-fc615828-4d40-11e7-807b-b988c19d5cdc.png)

Thanks @vaidap!

vaidap commented:
@YagoGG turns out we are not going with the bottom fish anymore for the design, so this isn't an issue anymore. thanks for reviewing!

vaidap commented:
added [WIP] as the artwork is still being finalised.

vaidap commented:
same fixes as for #5314. 
![image](https://user-images.githubusercontent.com/13666710/27791748-e15dd9a2-5ff6-11e7-992a-be3a9e8ce32e.png)

timabbott commented on static/html/5xx.html:
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html lang="en-US">
```
Is this change intentional?  In theory, we could have this page be translated.

vaidap replied:
added it to make it pass the accessibility tool-- will remove then!

timabbott commented on static/html/5xx.html:
```diff
@@ -8,34 +8,27 @@
          a copy of the file with no hash in the name.
          This file will not load in dev unless you set DEBUG = False. -->
-    <link href="/static/min/portico.css" rel="stylesheet">
+    <link href="/static/styles/portico.css" rel="stylesheet">
```
I assume we'll want to undo this hunk?  Maybe move it into its own "for testing" commit?

timabbott commented on static/html/5xx.html:
```diff
@@ -8,34 +8,27 @@
+        <div class="errorcontent">
+            <h1 class="lead">Internal server error</h1>
+            <p>The server has encountered some difficulties. Sorry about that!</p>
+            <p>The page will reload soon after service is restored.</p>
```
This should mention it's a Zulip server, at least.  I also feel "some difficulties" feels a bit unclear; I think in this case, it's fair to say "This Zulip server is currently experiencing some technical difficulties.".

Maybe the second sentence should read: "The page will reload automatically soon after service is restored."

vaidap commented:
thanks for the review @timabbott, i implemented your suggestions!

timabbott commented:
Merged as 0824314162a0162b282ad4df9d1caedb2f27e15d, after removing the unnecessary/unused `bottom.svg` file and `.for_you` CSS.  Thanks @vaidap!  

I think we may want to do a further tweak to this page to add some sort of Zulip heading to the top.  @brockwhittaker @rishig what are your thoughts?
